catalyst-gateway/Cargo.toml

```diff
@@ -45,7 +45,6 @@ dotenvy = "0.15"
 local-ip-address = "0.5.7"
 gethostname = "0.4.3"
 hex = "0.4.3"
-async-recursion = "1.0.5"
```

Comment:
are you sure this works without async recursion? Is the original recursive logic preserved?